File: doc/codes/csharp/examples/guide/custom-modulation.cs
using AUTD3;

namespace DocSamples.GuideCustomModulation;

internal static class Sample
{
    internal static void Run()
    {
        // ANCHOR: api
        var length = 10;
        var data = new ModulationBuffer(length);
        data[0] = 0xFF;

        new Modulation(SamplingConfig.Freq4k, data);
        // ANCHOR_END: api
    }
}

File: doc/codes/python/examples/guide/custom-modulation.py
from autd3.commands import Modulation
from autd3.value import SamplingConfig
from autd3_modulation import ModulationBuffer


def main() -> None:
    # ANCHOR: api
    length = 10
    data = ModulationBuffer(length)
    data[0] = 0xFF

    Modulation(SamplingConfig.FREQ_4K, data)
    # ANCHOR_END: api


if __name__ == "__main__":
    main()
